### ISSUE TITLE
[BUG] MacOS Share Extension needs CSRF cookie

### ISSUE THREAD

**Issue by zelig2** [LeoKlaus/Paperparrot#169]
**Describe the bug**
New install of paperparrot connecting to paperless-ngx server on Tailscale connected machine via http. 

**To Reproduce**
1. Go to local PDF in Finder 
2. Right click on PDF and select 'Share...'
3. Select 'Share'  NOTE: This should say Paperparrot not Share
4. Add descriptive title (or not)
5. See error

**Expected behavior**
A successful sending of the PDF to the paperless-ngx server.

**Screenshots**
<img width="512" alt="Screenshot 2023-12-27 at 3 12 30 PM" src="https://github.com/LeoKlaus/Paperparrot/assets/1232415/8096cd3c-320c-47ab-a63d-ab89da10ee6d">

**Environment(please complete the following information):**
 - Device: MacBook Pro M3 Pro
 - OS: Sonoma 14.1.1
 - Version 2.1.1
 - Server Version: Paperless-ngx 2.1.3
 - Tailscale 1.56.1

**Additional context**
Same behavior in the Share menu in Preview.

**Comment by LeoKlaus** [comment]
Hey zelig, thanks for reaching out!

I can't reproduce this on my Mac (connected via HTTPS over nginx, Paperparrot 2.1.1, ngx 2.1.2).

Do you have [PAPERLESS_URL](https://docs.paperless-ngx.com/configuration/#PAPERLESS_URL) set? Can you reach http://{instance-ip}:{port}/api/ in your browser without the CSRF error?

**Comment by zelig2** [comment]
I checked and I get a HTTP 200 response and everything looks good on my end in two different browsers. One I've logged in before and the other I have not. I get a good response from the API on my phone as well.  So that part all seems normal. 

<img width="565" alt="Screenshot 2023-12-27 at 5 25 29 PM" src="https://github.com/LeoKlaus/Paperparrot/assets/1232415/01f8cd83-7ba9-4457-b0bf-e0eecfedafe7">

**Comment by zelig2** [comment]
**solved**
I added another instance of my server, this time using the VPN address. That didn't work. Then I added the local address and it worked. After that I was able to start using the normal dns tailscale name in the share menu. Now that it's working I've deleted the duplicate servers and have it back to the single server. All is working normally now. There must have been some kind of weird cookie issue but it appears to be taken care of now.

**Comment by LeoKlaus** [comment]
Great, thanks for the update!

Do you remember in which order you added these instances to get to the issue in the first place?

**Comment by zelig2** [comment]
Yes, the first instance was firing up the app and loading my Tailscale domain name which is just "paperless". So that looks like http://paperless:8000. With the proper user credentials, Paperparrot loaded my library with no problem (only 6 items for the trial). I attempted to share a PDF on my Mac at that point but received the error shown at the beginning here. After that I paid for the full copy thinking it might be behind a paywall but received the same error. 

After that then added the Tailscale IP address which loaded in the app fine but gave the same error in the Finder. So I attempted to remove the initial http://paperless:8000 server. Strangely it wouldn't show it as removed when I tried that. I think that's because it was still loaded on the main screen for Paperparrot. Finally I added the local network IP address and switched the app over to use that account. At which I was able to remove the original Tailscale names version. The Finder worked at that point so I readded the Tailscale named version and it worked as well. I then removed all the other servers and I'm back to the original Tailscale named version with it all happy.
